lib/rust/profiling/src/macros.rs

```diff
@@ -0,0 +1,107 @@
+//! Utilities for easy profiling usage and definition.
+
+/// Create a Metadata struct that has the source location prepopulated via the `file!` and
+/// `line!` macros.
+#[macro_export]
+macro_rules! make_metadata {
+    ($profiling_level:expr, $interval_name:expr) => {
+        $crate::Metadata {
+            source:          $crate::SourceLocation { file: file!().to_string(), line: line!() },
+            profiling_level: $profiling_level.to_string(),
+            label:           $interval_name.to_string(),
+        }
+    };
+}
+
+/// Start measuring the interval of the given profiling level and name.
+#[macro_export]
+macro_rules! start_interval {
+    ($profiling_level:expr, $interval_name:expr) => {
+        $crate::mark_start_interval($crate::make_metadata!($profiling_level, $interval_name))
+    };
+}
+
+/// End measuring the interval of the given profiling level and name.
+#[macro_export]
+macro_rules! end_interval {
+    ($profiling_level:expr, $interval_name:expr) => {
+        let _ = $crate::mark_end_interval($crate::make_metadata!($profiling_level, $interval_name));
+    };
+}
+
+/// Measure an interval with the given profiling level and name that starts before the given
+/// expressions and ends after the given expressions.
+#[macro_export]
+macro_rules! measure_interval {
+        ($profiling_level:expr, $interval_name:expr, $($body:tt)*) => {
+             {
+                 $crate::start_interval!($profiling_level, $interval_name).release();
+                 let out = { $($body)* };
+                 $crate::end_interval!($profiling_level, $interval_name);
+                 out
```

Comment:
Shouldn't it be
```suggestion
                 let interval = $crate::start_interval!($profiling_level, $interval_name);
                 let out = { $($body)* };
                 $crate::IntervalHandle::end(interval);
                 out
```
?